storm-server/src/main/java/org/apache/storm/scheduler/multitenant/MultitenantScheduler.java

```diff
@@ -30,20 +32,22 @@
     private static final Logger LOG = LoggerFactory.getLogger(MultitenantScheduler.class);
     protected IConfigLoader configLoader;
     private Map<String, Object> conf;
+    private SchedulerConfigCache<Map<String, Number>> schedulerConfigCache;
 
     @Override
     public void prepare(Map<String, Object> conf, StormMetricsRegistry metricsRegistry) {
         this.conf = conf;
         configLoader = ConfigLoaderFactoryService.createConfigLoader(conf);
-
+        schedulerConfigCache = new SchedulerConfigCache<>(conf, this::loadConfig);
```

Comment:
Since SchedulerConfigCache is now just a wrapper around loadConfig() method, can the class be removed and loadConfig() be called directly at the beginning of schedule?

Reply:
There are currently two schedulers (ResourceAwareScheduler, MultitenantScheduler) need to have this logic. And maybe in the future a new scheduler or other existing schedulers will want to have configs too.  I keep this class to wrap the logic and to avoid repeat the code in schedulers.